sources/web/datalab/polymer/components/file-browser/file-browser.ts

```diff
@@ -285,6 +299,49 @@ class FileBrowserElement extends Polymer.Element implements DatalabPageElement {
     this._loadStartupPath(fileId);
   }
 
+  async _fileManagerTypeListChanged() {
+    if (!this.fileManagerTypeList) {
+      const settings = await SettingsManager.getAppSettingsAsync();
+      // Fall back to Jupyter if settings are somehow outdated.
+      const types = settings.supportedFileBrowserSources || ['jupyter'];
+      this.fileManagerTypeList = types.map((source) =>
+          FileManagerFactory.fileManagerNameToType(source));
+    }
+
+    const menu = this.$.fileSourcesDropdown as HTMLDivElement;
+    // Build and add buttons to the switcher dropdown menu
+    Utils.deleteAllChildren(menu);
+
+    this.fileManagerTypeList.forEach((type) => {
+      const config = FILE_MANAGER_CONFIG.get(type);
+      if (!config) {
+        throw new Error('Unknown FileManagerType: ' + type.toString());
+      }
+      const btn = document.createElement('paper-button');
+      btn.classList.add('toolbar-button');
+      btn.addEventListener('click', () => {
+        if (this.fileManagerType !== type) {
+          this.fileManagerType = type;
+          this._fileManager = FileManagerFactory.getInstanceForType(type);
+          this.fileId = '';
```

Comment:
Might be nice to store the last location used with this file manager, so that if you switch from, say My Drive to Shared Drive and back to My Drive, you will end up in the same place in My Drive as when you left it. But that could be in a future PR.

Reply:
Yes I thought about that. The `_loadStartupPath` method should take care of loading the last accessed path for the selected file manager.